openedx_ledger/models.py

```diff
@@ -315,6 +334,64 @@ class Meta:
     history = HistoricalRecords()
 
 
+class ExternalTransactionReference(TimeStampedModel):
+    """
+    Model of references to transactions from an external system and their associated Transaction objects.
+    """
+
+    class Meta:
+        """
+        Metaclass for ExternalTransactionReference.
+        """
+
+        unique_together = [('external_reference_id', 'external_reference_type')]
+        index_together = [('external_reference_id', 'external_reference_type')]
+
+    transaction = models.ForeignKey(
+        Transaction,
+        related_name='external_reference',
+        null=True,
+        on_delete=models.CASCADE,
+        help_text=(
+            "The Transaction to which this external reference is associated."
+        ),
+    )
+    external_reference_id = models.CharField(
+        max_length=255,
+        blank=False,
+        null=False,
+        primary_key=True,
+        db_index=True,
+        help_text=(
+            "The name of the external reference.  This is typically the name of the system that the reference is "
+            "associated with."
+        ),
+    )
+    external_reference_type = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+        # Since null=True we do not need a default choice.  Furthermore, a default doesn't make sense anyway because
+        # there's no reasonable heuristic to guess the type before runtime.
+        choices=TransactionReferenceTypeChoices.CHOICES,
```

Comment:
We should take this opportunity to just re-do the choices to something closer to reality, e.g.:
```python
    GEAG_ALLOCATION_ID = 'geag_allocation_id'
    CHOICES = (
        (GEAG_ALLOCATION_ID, 'GEAG Allocation ID'),
    )
```